src/Microsoft.Azure.WebJobs.Extensions.Kafka/Listeners/KafkaListener.cs

```diff
@@ -106,6 +123,8 @@ public Task StartAsync(CancellationToken cancellationToken)
 
             consumer.Subscribe(this.listenerConfiguration.Topic);
 
+            this.topicScaler = new KafkaTopicScaler<TKey, TValue>(this.listenerConfiguration.Topic, this.consumerGroup, scaleMonitorDescriptor, consumer, new AdminClientConfig(GetConsumerConfiguration()), this.logger);
```

Comment:
It looks like all the actual logic is done in KafkaTopicScaler. Recommend then that this listener class just implement IScaleMonitorProvider and return the topic scaler. KafkaTopicScaler then implements IScaleMonitor. See the EventHub implementation [here](https://github.com/Azure/azure-functions-eventhubs-extension/blob/daab3ae8a8e08dd971b8f9b19d52f09ef44af081/src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs#L24).